backend/app/api/endpoints/adapter/tasks.py

```
@@ -280,6 +284,88 @@ def get_task_skills(
+    current_user: User = Depends(security.get_current_user),
+    db: Session = Depends(get_db),
+):
+    """Generate prompt draft as SSE stream events."""
+
+    try:
+        # Pre-check to return 4xx before streaming starts.
+        prompt_draft_service.validate_prompt_draft_context(
+            db=db, task_id=task_id, current_user=current_user, model=request.model
+        )
+    except prompt_draft_service.PromptDraftTaskNotFoundError:
+        raise HTTPException(status_code=404, detail="Task not found")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)